src/scheduler.rs

```diff
@@ -65,32 +68,62 @@ async fn scheduler_loop(
                 checks_scheduled,
                 "scheduler.tick_execution_complete"
             );
+            tick
         });
+
+        tick_complete_tx
+            .send(tick_complete_join_handle)
+            .expect("Failed to queue join handle for tick completion");
     };
 
+    let in_order_tick_processor_join_handle = tokio::spawn(async move {
+        while let Some(tick_complete_join_handle) = tick_complete_rx.recv().await {
+            // XXX: This task guarantees that we can process ticks IN ORDER upon tick execution.
+            // This waits for the tick to complete before waiting on the next tick to complete.
+            let tick = tick_complete_join_handle
+                .await
+                .expect("Failed to recieve completed tick");
+
+            tracing::debug!(tick = %tick, "scheduler.tick_execution_complete_in_order");
+        }
+    });
+
     while !shutdown.is_cancelled() {
         let interval_tick = interval.tick().await;
         let tick = Tick::from_time(start + interval_tick.duration_since(instant));
         schedule_checks(tick);
     }
+
+    // Wait for all dispatch ticks to have completed
+    drop(tick_complete_tx);
```

Comment:
This basically just closes the channel? Does that end up causing `.recv` to return None so that the loop exits?

Reply:
yeah exactly

Reply:
actually I think I can make it more explicit.

Reply:
nvm this is the best way